src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509.cpp

```diff
@@ -0,0 +1,478 @@
+        {
+            if (dataFormat == kSecFormatPKCS7)
+            {
+                return PAL_Pkcs7;
+            }
+
+            if (dataFormat == kSecFormatPKCS12)
+            {
+                return PAL_Pkcs12;
+            }
+
+            printf("DataFormat: %d\n", dataFormat);
+        }
+
+        printf("ActualType was not certificate: %d\n", actualType);
+    }
+    else if (osStatus != errSecUnsupportedFormat)
+    {
+        printf("Call 1: %d\n", osStatus);
+    }
+
+    actualType = kSecItemTypeAggregate;
+    dataFormat = kSecFormatPKCS7;
+
+    osStatus = SecItemImport(cfData, nullptr, &dataFormat, &actualType, 0, nullptr, nullptr, nullptr);
+
+    printf("Call 2: osStatus (%d), dataFormat (%d), actualType (%d)\n", osStatus, dataFormat, actualType);
+
+    if (osStatus == noErr)
+    {
+        if (actualType == kSecItemTypeAggregate && dataFormat == kSecFormatPKCS7)
+        {
+            return PAL_Pkcs7;
+        }
+
+        printf("ActualType (%d), dataFormat (%d)\n", actualType, dataFormat);
+    }
+    else if (osStatus != errSecUnsupportedFormat)
+    {
+        printf("Call 2: %d\n", osStatus);
+    }
+
+    actualType = kSecItemTypeAggregate;
+    dataFormat = kSecFormatPKCS12;
+
+    osStatus = SecItemImport(cfData, nullptr, &dataFormat, &actualType, 0, nullptr, nullptr, nullptr);
+
+    printf("Call 3: osStatus (%d), dataFormat (%d), actualType (%d)\n", osStatus, dataFormat, actualType);
+
+    if (osStatus == noErr || osStatus == errSecPassphraseRequired)
+    {
+        if (actualType == kSecItemTypeAggregate && dataFormat == kSecFormatPKCS12)
+        {
+            return PAL_Pkcs12;
+        }
+
+        printf("ActualType (%d), dataFormat (%d)\n", actualType, dataFormat);
+    }
+    else if (osStatus != errSecUnsupportedFormat)
+    {
+        printf("Call 3: %d\n", osStatus);
+    }
+
+    return PAL_X509Unknown;
+}
+
+static void ForceLoadPrivateKey(SecIdentityRef identity)
+{
+    // Exporting the key will cause it to be loaded out of the keychain before
+    // we delete the keychain.  And since the same object is kept alive across
+    // multiple calls to SecIdentityCopyPrivateKey, this makes it usable to the
+    // rest of the process
+    SecKeyRef privateKey = nullptr;
+    OSStatus local = SecIdentityCopyPrivateKey(identity, &privateKey);
+
+    if (local == noErr)
+    {
+        SecExternalFormat dataFormat = kSecFormatWrappedPKCS8;
+        SecItemImportExportKeyParameters keyParams = {};
+        keyParams.version = SEC_KEY_IMPORT_EXPORT_PARAMS_VERSION;
+        keyParams.passphrase = CFSTR("ExportPassphrase");
+
+        CFDataRef exported = nullptr;
+
+        SecItemExport(privateKey, dataFormat, 0, &keyParams, &exported);
+
+        CFRelease(keyParams.passphrase);
+
+        if (exported != nullptr)
+        {
+            CFRelease(exported);
+        }
+    }
+
+    if (privateKey != nullptr)
+    {
+        CFRelease(privateKey);
```

Comment:
It forced it out of the keychain and into memory.  The release here matches SecIdentityCopyPrivateKey having up-ref'd it.

But this whole function is bogus, it was from when I was trying to do

TempKeyChain
ReadPFX => identity
ForceLoadKey
DeleteTempKeyChain
return identity

But that caused problems, so now I have the less-foolproof but more error-proof delayed deletion on the keychain.